formatTest/typeCheckedTests/input/sequences.re

```diff
@@ -79,3 +79,21 @@ let thirdFieldPunned = {
   c
 };
 let singlePunAcceptedIfExtended = {...firstFieldPunned, a};
+
+
+let opt = (value, fn) => switch value {
+  | None => None
+  | Some(x) => fn(x)
+};
+
+let opt2 = (values, fn) => switch values {
+  | (Some(a), Some(b)) => fn((a, b))
+  | _ => None
+};
+
+let _ = {
+  let!opt x = Some(10);
```

Comment:
Can you also add tests for the sugar outside of let bindings, e.g. structure_items, Pstr_eval?

Reply:
Let bindings are the only place it's allowed. structure_items will throw an error (because there's no continuation, it doesn't really make sense).
I think I also want to disallow extensions... although I could be convinced